Source/artifacts/EventTypes.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Guid } from '@dolittle/rudiments';
+import { Constructor } from '@dolittle/types';
+import { UnknownType } from './UnknownType';
+import { EventTypeMap } from './EventTypeMap';
+import { IEventTypes } from './IEventTypes';
+import { EventType } from './EventType';
+import { UnknownEventType } from './UnknownEventType';
+import { EventTypeId } from './EventTypeId';
+import { UnableToResolveEventType } from './UnableToResolveEventType';
+import { CannotHaveMultipleEventTypesAssociatedWithType } from './CannotHaveMultipleEventTypesAssociatedWithType';
+import { CannotHaveMultipleTypesAssociatedWithEventType } from './CannotHaveMultipleTypesAssociatedWithEventType';
+
+/**
+ * Represents an implementation of {@link IEventTypes}
+ */
+export class EventTypes implements IEventTypes {
+    /**
+     * Initializes a new instance of {@link Artifacts}
+     * @param {Map<Function, Artifact>?} associations Known associations
+     */
+    constructor(private _associations: EventTypeMap<Constructor<any>> = new EventTypeMap()) {
+    }
+    /** @inheritdoc */
+    hasTypeFor(input: EventType): boolean {
+        for (const associatedArtifact of this._associations.keys()) {
+            if (this.eventTypesEquals(input, associatedArtifact)) return true;
+        }
+        return false;
```

Comment:
Should be able to use `_associations.has(...)` here?

Reply:
Yep